R/calcualte_age.R

```diff
@@ -21,78 +26,88 @@
-  tmp_age <- remainder_days <- NULL
+    choices = c("years", "months", "weeks", "days"),
+    null.ok = FALSE
+  )
+  checkmate::assert_choice(target_column,
+    choices = colnames(data),
+    null.ok = TRUE
+  )
+  end_date <- checkmate::assert_date(
+    as.Date(end_date),
+    any.missing = FALSE, len = 1L, null.ok = TRUE
+  )
+
+  # check age column name
+  checkmate::assert_string(age_column_name)
+
+  # check age remainder and match to options
+  # NOTE: "year" is not an option, but a remainder in "weeks" could be specified
+  # for `age_in = "weeks"` - expect user will make common sense decisions
+  age_remainder_unit <- match.arg(age_remainder_unit)
+  # get age remainder column name
+  age_remainder_colname <- sprintf("remainder_%s", age_remainder_unit)
 
   # replace missing data characters with NA
   data <- replace_missing_values(data, target_column,
-                                 na_strings = na_strings)
+    na_strings = na_strings
+  )
 
   # standardize the input data if required
+  # NOTE: define `else` case or check that target column is a `Date`
   if (!lubridate::is.Date(data[[target_column]])) {
     # the error_tolerance = 0.0 because target_column is explicit
     data <- standardize_dates(data, target_column,
-                              format          = NULL,
-                              timeframe       = NULL,
-                              error_tolerance = 0.0)
+      format          = NULL,
+      timeframe       = NULL,
+      error_tolerance = 0.0
+    )
   }
 
-  # calculate the age
-  res <- switch(age_in,
-    years = data %>%
-      dplyr::mutate(age_years = round((data[[target_column]] %--% end_date)
-      %/% lubridate::years(1L))),
-    months = data %>%
-      dplyr::mutate(tmp_age = lubridate::as.period(end_date -
-                                                     data[[target_column]])) %>%
-      dplyr::mutate(
-        age_months = tmp_age %/% months(1L, abbreviate = FALSE),
-        remainder_days = (tmp_age %% months(1L, abbreviate = FALSE)) %/%
-          lubridate::days(1L)
-      ) %>%
-      dplyr::select(-tmp_age),
-    days = data %>%
-      dplyr::mutate(tmp_age = lubridate::as.period(end_date -
-                                                     data[[target_column]])) %>%
-      dplyr::mutate(
-        age_days = tmp_age %/% lubridate::days(1L)
-      ) %>%
-      dplyr::select(-tmp_age),
-    weeks = data %>%
-      dplyr::mutate(tmp_age = lubridate::as.period(end_date -
-                                                     data[[target_column]])) %>%
-      dplyr::mutate(
-        age_weeks = tmp_age %/% lubridate::weeks(1L),
-        remainder_days = (tmp_age %% lubridate::weeks(1L))
-        %/% lubridate::days(1L)
-      ) %>%
-      dplyr::select(-tmp_age)
+  # switch divisor based on requested unit
+  # NOTE: no default case defined, add a default case?
+  divisor_age <- switch(age_in,
+    years = lubridate::years(1L),
+    months = months(1L), # from base
```

Comment:
Could you leave a quick note as to why months had to come from base? I had to go check 😅 .

Reply:
Added in https://github.com/epiverse-trace/cleanepi/pull/83/commits/0c647d294f55ee3e17cb679950cdd28390da82f5

Reply:
On the contrary, it is {lubridate} which [provides the `months.numeric()` method](https://github.com/tidyverse/lubridate/blob/e0b50c1759fe35e90a094c012e0c2ce60d47500d/R/periods.r#L541), but that is accessed by calling simply `base::months()` on a numberic.

Reply:
Seeing the comment in the snippet above remains me that the default could be when the remainder is sent as `digits`.
what do you think?

Reply:
Yes could be, maybe something for issue #90